interp/test.go

```diff
@@ -200,6 +204,24 @@ func (r *Runner) unTest(ctx context.Context, op syntax.UnTestOperator, x string)
 		return r.lookupVar(x).Kind == expand.NameRef
 	case syntax.TsNot:
 		return x == ""
+	case syntax.TsUsrOwn, syntax.TsGrpOwn:
+		if runtime.GOOS == "windows" {
+			panic(fmt.Sprintf("unhandled unary test op: %v", op))
+		}
+		fi, err := os.Stat(x)
+		if err != nil {
+			return false
+		}
+		u, err := user.Current()
+		if err != nil {
+			return false
+		}
+		if op == syntax.TsUsrOwn {
```

Comment:
What about this ?

```suggestion
stat, ok :=  fi.Sys().(*syscall.Stat_t)
if !ok {
      return false
}
   
if op == syntax.TsUsrOwn {
       return strconv.Itoa(uid) == stat.Uid
} 
return return strconv.Itoa(gid) == stat.Gid
```

Reply:
This wouldn't build either, though. The problem is that under Windows the `syscall.Stat_t` type doesn't even exist, not whether the type assertion would succeed or fail.

I think I thought/hoped that the `runtime.GOOS` check would be resolved statically at compile time and the compiler would ignore the rest of the `case`, but alas, that was not, uh, the case. :) In reality, I think the GOOS check *is* resolved statically at compile time, but the rest of the code still needs to compile.

Reply:
ah ok, the need was different than I thought